orttraining/orttraining/core/graph/allreduce_optimizer_graph_builder.cc

```diff
@@ -92,61 +92,29 @@ Status AllreduceOptimizerGraphBuilder::AddHorovodAllReduceForGradients(std::vect
 
 static Status AddNcclAllReduceForGradients(
     std::vector<ArgDef>& gradient_argdefs,
-    ArgDef& fused_gradient_argdef,
-    GraphAugmenter::GraphDefs& graph_defs,
-    ArgDef& fused_allreduce_output) {
-  fused_allreduce_output = ArgDef(fused_gradient_argdef.name + "AllReduce_Out", fused_gradient_argdef.type_proto);
-
-  // Add NCCL Allreduce node.
-  graph_defs.AddNodeDefs({NodeDef(OpDef{"NcclAllReduce", kMSDomain, 1},
-                                  {fused_gradient_argdef},
-                                  {fused_allreduce_output},
-                                  NodeAttributes(),
-                                  "NcclAllReduce")});
-
-  std::vector<ArgDef> view_inputs(gradient_argdefs.size() + 1);
-  view_inputs[0] = fused_allreduce_output;
-
-  for (size_t i = 0; i < gradient_argdefs.size(); i++) {
-    ArgDef& gradient_shape = view_inputs[i + 1];
-    gradient_shape = ArgDef(gradient_argdefs[i].name + "_Shape");
-
-    graph_defs.AddNodeDefs({NodeDef("Shape",
-                                    {gradient_argdefs[i]},
-                                    {gradient_shape},
-                                    NodeAttributes(),
-                                    gradient_shape.name)});
-  }
+    std::vector<ArgDef>& fused_gradient_argdef,
+    GraphAugmenter::GraphDefs& graph_defs) {
 
   std::vector<ArgDef> allreduce_outputs(gradient_argdefs.size());
   for (size_t i = 0; i < gradient_argdefs.size(); i++) {
     TypeProto* allreduced_gradient_type_proto = graph_defs.CopyTypeProto(gradient_argdefs[i]);
     allreduced_gradient_type_proto->mutable_tensor_type()->set_elem_type(
-        fused_gradient_argdef.type_proto->tensor_type().elem_type());
+        fused_gradient_argdef[0].type_proto->tensor_type().elem_type());
```

Comment:
Can we get rid of `fused_gradient_argdef` now?

Reply:
Its just the name, its not fused. I have renamed it.

---
In reply to: [505116367](https://github.com/microsoft/onnxruntime/pull/4443#discussion_r505116367) [](ancestors = 505116367)